src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+                    if (identityClaim != null)
+                    {
+                        this.remoteIdentity = EndpointIdentity.CreateIdentity(identityClaim);
+                    }
+                    this.sessionTokenIdentifier = settings.IssuedSecurityTokenParameters.CreateKeyIdentifierClause(sessionToken, SecurityTokenReferenceStyle.External);
+                    this.standardsManager = settings.SessionProtocolFactory.StandardsManager;
+                }
+
+                public string Id
+                {
+                    get
+                    {
+                        return this.securityContextTokenId.ToString();
+                    }
+                }
+
+                public EndpointIdentity RemoteIdentity
+                {
+                    get
+                    {
+                        return this.remoteIdentity;
+                    }
+                }
+
+                public void WriteSessionTokenIdentifier(XmlDictionaryWriter writer)
+                {
+                    this.standardsManager.SecurityTokenSerializer.WriteKeyIdentifierClause(writer, this.sessionTokenIdentifier);
+                }
+
+                public bool TryReadSessionTokenIdentifier(XmlReader reader)
+                {
+                    if (!this.standardsManager.SecurityTokenSerializer.CanReadKeyIdentifierClause(reader))
+                    {
+                        return false;
+                    }
+                    SecurityContextKeyIdentifierClause incomingTokenIdentifier = this.standardsManager.SecurityTokenSerializer.ReadKeyIdentifierClause(reader) as SecurityContextKeyIdentifierClause;
+                    return incomingTokenIdentifier != null && incomingTokenIdentifier.Matches(this.securityContextTokenId, null);
+                }
+            }
+
+
+            //Renamed SecurityReplySessionChannel => SecurityReplySessionServiceChannelDispatcher (implementd IServiceChannelDispatcher)
+            public class SecurityReplySessionServiceChannelDispatcher : ServerSecuritySimplexSessionChannel, IServiceChannelDispatcher //, IReplySessionChannel 
+            {
+                private IReplyChannel incomingChannel;
+                public SecurityReplySessionServiceChannelDispatcher(
+                    SecuritySessionServerSettings settings,
+                    //  IServerReliableChannelBinder channelBinder,
+                    SecurityContextSecurityToken sessionToken,
+                    object listenerSecurityState, SecurityListenerSettingsLifetimeManager settingsLifetimeManager
+                    , IChannel channel, EndpointAddress address)
+                    : base(settings,
+                          // channelBinder,
+                          sessionToken, listenerSecurityState, settingsLifetimeManager, address)
+                {
+                    this.incomingChannel = (IReplyChannel) channel;
+                    base.OpenAsync(ServiceDefaults.OpenTimeout);
```

Comment:
You are calling an async method from a constructor. Instead have the code which constructs an instance of this class call OpenAsync on this instance.